### PR TITLE
Create verse and verses queries

### LINKED ISSUE #26: verses GraphQL query
Create GraphQL query to return all verses

Request should be:
query {
  verses {
     id
    trackId
    userId
    audioPath
    user { userType }
    votes { voteType }
    track { trackType }
  }
}

Response should be:
{
  "data": {
    "verses": [
       {
          "id": <string>,
          "trackId": <string>,
          "userId": <string>,
          "audioPath": <string>,
          "user": { userType }
          "votes": [ { voteType } ]
          "track": { tracktype }
       },
       { .... }
     ]
  }
}

### LINKED ISSUE #30: Single Verse GraphQL Query
- [ ] Create graphql query to return information for a single verse from the database
- [ ] create type for verse query

**request**
query {
  verse(id: ID) {
    id
    trackId
    userId
    audioPath
    user { userType }
    votes { voteType }
    track { trackType }
  }
}

**response**
{
  "data": {
     "verse": {
       "id": <string>,
       "trackId": <string>,
       "userId": <string>,
       "audioPath": <string>,
       "user": { usertype }
       "votes": [ { votetype) ] <- array of vote objects
       "track": { trackType }
     }
  }
}

### PR DESCRIPTION
**Pull request by ETBassist**
- Create test for single verse query
- Create query to get a single verse
- Create test for getting all verses through graphql
- Create query to get all verses through GraphQL

## Types of Changes

This Pr creates queries to get all verses and a single verse by id

This PR closes #26 and closes #30

- [ ] Bug fix (non-breaking change which fixes an issue): 

- [x] New feature (non-breaking change which adds functionality):

- [ ] Breaking change (fix or feature that would cause existing functionality to change):

- [ ] My change requires a change to the documentation.

- [ ] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

### CODE REVIEW

File: spec/factories.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@
     competition_id { rand(100) }
     user_id { rand(100) }
     audio_path { Faker::Internet.url }
+    title { Faker::Books::Lovecraft.deity }
```
Comment (SageOfCode) [inline spec/factories.rb:21]: Good call on this library

File: app/graphql/types/verse_type.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Types
```
Comment (SageOfCode) [inline app/graphql/types/verse_type.rb:1]: I think we also had to create this type, may lead to a conflict

**Review by SageOfCode — APPROVED** [review]
Great work team, looks good to me.

File: app/graphql/types/verse_type.rb (hunk 2 above)
Comment (the-color-bliu) [reply app/graphql/types/verse_type.rb:1]: We definitely did, although the conflict may just be how the fields are ordered

File: app/graphql/types/query_type.rb [hunk 3]
```diff
@@ -11,5 +11,19 @@ class QueryType < Types::BaseObject
...
+      Verse.find(id)
+    end
+
+    field :verses, [Types::VerseType], null: false 
+
+    def verses
+      Verse.all
+    end
   end
 end
```
Comment (the-color-bliu) [inline app/graphql/types/query_type.rb:29]: I'm liking the look of these query types

**Review by the-color-bliu — APPROVED** [review]
Looks great, merge approved!

**Pull request merged**